### DISCUSSION TITLE
Conflicts with the 'debug' packages and best steps forward?

### DISCUSSION THREAD

**Discussion by dmoonfire** [atom/atom#22121] [General]
So, I was looking into the conflict I have between two different versions of "debug". In specific, I switched `spell-check` to use "debug" which didn't exist as a package in APM, so it fell back to the NPM version which gave me the ability to set a local storage value that would give me insight into how the spell checking was figuring out the answers.

A few months later, atom community created "debug" which then spell-check started using (as far as I can), so I had a conflict.

Today, I was looking at the API to see if that debug was going to work better but I noticed the [project](https://github.com/atom-community-archive/debug) was archived but there was no corresponding new project in the new community. On the original project, there is an [issue](https://github.com/atom-community-archive/debug/issues/29) about unpublishing or creating a redirect but since the project was archived, I couldn't answer.

The question is, what is my best step forward?

- Create a mirror copy of `debug` so I can reference it to a new name?
- Teach APM how to identify packages that need to come from NPM verses APM?
- Do something with the APM version of `debug` that would let me use the NPM version?
- Write a different debug library?
- Find a different way to solve the problem? (see below)

Overall, the idea was to be able to have a user turn on a flag (or the new "debug" flag that is dropping in 77) and have it dump something that could then be copy/pasted into a issue ticket to help me investigate? A lot of the problems seem to be environment specific (operating system, languages installed, how the languages are installed) and something I haven't been easily been able to identify beyond questions. Having a trace helps with that.

Any help would be greatly appreciated.

**Comment by UziTech** [comment]
NPM and APM packages are completely different. For a package to use an NPM package they can use `require` or `import`. For a package to use another atom package they would have to use the [Service API](https://flight-manual.atom.io/behind-atom/sections/interacting-with-other-packages-via-services/). So there should be no conflict with naming.

**Reply by dmoonfire** [reply]
Thank you. Apparently I went down the wrong path in trying to figure it out.

```
this.log = debug('spell-check:locale-checker').extend(locale);
this.log.log = console.log.bind(console);
```

The fix was that I needed to bind to the console. Not sure why that changed because it was working when I had the PR submitted, but then it wsan't. Using the above bind seems to have fixed it and I got debugging messages again.

Thank you!
